src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -1,192 +1,139 @@
+                frontRightPositionController, rearLeftPositionController, rearRightPositionController);
 
+        reset();
     }
-
-    @Override 
-    public void periodic(){
-        logger.log("Front Left Encoder Ticks", e_frontLeft.getPosition());
-        logger.log("Front Right Encoder Ticks", e_frontRight.getPosition());
-        logger.log("Rear Left Encoder Ticks", e_rearLeft.getPosition());
-        logger.log("Rear Right Encoder Ticks", e_rearRight.getPosition());
+    public void setSpeedMode(){
+        speedSettings.configureSparkMax(frontLeftSpark);
+        speedSettings.configureSparkMax(frontRightSpark);
+        speedSettings.configureSparkMax(rearLeftSpark);
+        speedSettings.configureSparkMax(rearRightSpark);        
     }
-
-    public void reset(){
-        pc_frontLeft.resetPosition();
-        pc_frontRight.resetPosition();
-        pc_rearLeft.resetPosition();
-        pc_rearRight.resetPosition();
-        logger.log("Clicks per rotation", e_rearRight.getCountsPerRevolution());
+    public void setPositionMode(){
+        positionController.configureAll();
     }
     
+    public EncoderValues getEncoderValues(){
+       return new EncoderValues(frontLeftEncoder.getPosition(), 
+                rearLeftEncoder.getPosition(), 
+                frontRightEncoder.getPosition(), 
+                rearRightEncoder.getPosition());
+    }
+   
+    @Override 
+    public void customPeriodic(RobotPose rp, FieldPose fp) {
+        
+        logger.log("Front Left Encoder Ticks", frontLeftEncoder.getPosition());
+        logger.log("Front Right Encoder Ticks", frontRightEncoder.getPosition());
+        logger.log("Rear Left Encoder Ticks", rearLeftEncoder.getPosition());
+        logger.log("Rear Right Encoder Ticks", rearRightEncoder.getPosition());
+        
+    }
 
-    public void updatePose(PositionReader pose){
-        poseGen.updateFromOfficialPose(pose);
-        poseGen.updatePose();
+    private void localReset(){
+        frontLeftPositionController.resetPosition();
+        frontRightPositionController.resetPosition();
+        rearLeftPositionController.resetPosition();
+        rearRightPositionController.resetPosition();
+        logger.log("Clicks per rotation", rearRightEncoder.getCountsPerRevolution());
     }
-    public void drive(DriveInstruction di) {
-        m_robotDrive.arcadeDrive(di.getFoward(), di.getRotation());
+    public Command reset(){
+        return new ResetPositionCommand(this);
     }
-    public EncoderPoseGenerator getEncoderPoseGenerator(){
-        return this.poseGen;
+
+    public void drive(DriveInstruction di) {
+        robotDrive.arcadeDrive(di.getFoward(), di.getRotation());
     }
-}
+
+    public class ResetPositionCommand extends SingleShotCommand {
+
+        public ResetPositionCommand(BaseSubsystem bs) {
+            super(bs);
+        }
+
+        @Override
+        public void doCommand() {
+            localReset();
```

Comment:
why not just do the code of localReset right here? it would make sense to have a method if its a lot of code, but in this case, its not a lot